src/fastga/models/aerodynamics/components/figure_digitization.py

```diff
@@ -2400,16 +2093,26 @@ def cn_r_drag_effect(static_margin, sweep_25, aspect_ratio) -> float:
 
         return float(drag_effect)
 
-    @staticmethod
-    def interpolate_database(database, tag_x: str, tag_y: str, input_x: float):
-        database_x = database[tag_x]
-        database_y = database[tag_y]
-        errors = np.logical_or(np.isnan(database_x), np.isnan(database_x))
-        database_x = database_x[np.logical_not(errors)].tolist()
-        database_y = database_y[np.logical_not(errors)].tolist()
-
-        output_y = interpolate.interp1d(database_x, database_y)(
-            np.clip(input_x, min(database_x), max(database_x))
-        )
 
-        return output_y
+def interpolate_database(database, tag_x: str, tag_y: str, input_x: float):
+    database_x, database_y = filter_nans(database, tag_x, tag_y)
+
+    output_y = interpolate.interp1d(database_x, database_y)(
+        np.clip(input_x, min(database_x), max(database_x))
+    )
+
+    return output_y
+
+
+def filter_nans(database: pd.DataFrame, tag_x: str, tag_y: str):
+    """
+    Utility function to jointly filter out NaN in the database with the selected tags.
+    """
+
+    database_x = database[tag_x]
+    database_y = database[tag_y]
+    errors = np.logical_or(np.isnan(database_x), np.isnan(database_x))
+    database_x = database_x[np.logical_not(errors)].tolist()
+    database_y = database_y[np.logical_not(errors)].tolist()
```

Comment:
I can suggest the use of pandas utility function `[Dataframe.dropna()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dropna.html)`. It would look something like:

`subset = database.dropna(subset=[tag_x, tag_y])`
`database_x = subset[tag_x].to_list()`
`database_y = subset[tag_y].to_list()`

tag_x and tag_y could be replaced by a list of names and the function return a list as well for more general use.
I haven't tested though.

Reply:
Thank you for the suggestion, it is quicker and would be more generic. Do you want a re-review once this is implemented ?

Reply:
Correction: it's actually not faster though not by a significant margin but it still is more generic in my opinion so I suggest we keep it @esnguyenvan

Reply:
The way you implement it is good for me. Sad that it's not fastest but it's more generic and cleaner.

Reply:
I'd say we're good for this PR, thank for the effort to remove the last codacy warning